packages/gatsby-transformer-mdx-introspection/README.md

```diff
@@ -40,31 +45,108 @@ query MyQuery {
 }
 ```
 
-Please not the lowercase `apitype` in the query - the names are normalized to lowercase by the DOM parser used.
-
-Example response:
+### Example response:
 
 ```json
 {
   "data": {
     "allComponentInMdx": {
       "nodes": [
         {
+          "component": "ApiType",
           "attributes": [
             {
-              "name": "apikey",
+              "name": "apiKey",
               "value": "test"
             },
             {
               "name": "type",
               "value": "OutOfOrderPropertiesTestType"
             }
           ],
-          "page": {
-            "fields": {
-              "slug": "/api/types/"
+          "mdx": {
+            "file": {
+              "relativePath": "api/types/index.md"
             }
           }
         },
-...
+        ...
+      ]
+    }
+  }
+}
+```
+
+## Options
+
+**`cleanWhitespace`** [boolean] (optional, defaults to true)
+
+Whether to collapse/trim whitespace in JSX snippets and string literals
+
+**`removeMdxCompilationArtifacts`** [boolean] (optional, defaults to true)
+
+Whether to remove attributes that are (usually) artifacts of MDX compilation (`mdxType` and `parentName`)
+
+**`shouldIndexNode`** [(node) ⇒ boolean] (optional, defaults to () ⇒ true)
+
+Predicate function used as a performance escape hatch to filter MDX files that get parsed/indexed. Use if not all MDX files need to be indexed and build times are prohibitive
+
+**`tagWhitelist`** [array\<string | RegExp>] (optional, defaults to `['a', /^h[1-6]\$/]`)
+
+JSX components that generate Gatsby data nodes in the final output (other nodes still appear as children in `tree` and their children can generate nodes)
+
+##### Default value
+
+By default, the plugin only generates Gatsby data nodes for the following elements:
+
+- `a`
+- `h1`
```

Comment:
I wonder whether the headings are a good default - the heading structure is already available from the MDX plugin GraphQL and if you can use that you should:

```graphql
query MdxHeadings {
  allMdx {
    nodes {
      headings {
        depth
        value
      }
    }
  }
}
````

If the idea behind these defaults is to point out that by using this plugin instead of the mdx plugin's built-in you can cover more or different cases we should explain that difference to point out advantages.

Reply:
@jazevedo620  Do you have an opinion here?

Reply:
I was thinking about and I didn't have a strong attachment to the current defaults of `a` and `h1-6`, they just seemed like a useful starting point to introspect in documentation. I had forgot about the normal MDX plugin's header querying functionality, but the default value isn't designed to really be used that much.

In that case, it might be better to have the whitelist be a required option for the plugin to work, and it could warn to console/panic on build if it doesn't detect that it has been set.

Reply:
@nkuehn Do you have an opinion on whether the default should be changed to an empty whitelist or whether it should kept as is?

Reply:
I suggest to document that it's a required parameter and leave the default empty.   We'll have happier users due to the performance gain. 

README should explain how to put a wildcard to first introspect everything and then put what you were looking for.